### PR TITLE
adding read function by source

### PR DESCRIPTION
**Pull request by caseyjlaw**

### CODE REVIEW

File: lwa_antpos/reading.py [hunk 1]
```diff
@@ -13,6 +13,24 @@
 # "data/LWA-352 Antenna Positions & System Status.xlsx"  # old name
 
 
+def read_antpos(source='yaml'):
```
Comment (yupinghuang) [inline lwa_antpos/reading.py]: I'd recommend the docs specifying that the default is typically a good choice.

and either use `source=None` to denote default (aka let casey choose how I get them) or `default`. may be a lil clearer that way.

Make `read_antpos_xxx` start with underscore to denote privateness.

File: lwa_antpos/reading.py [hunk 2]
```diff
@@ -63,3 +69,16 @@ def read_antpos_yaml(filename=antposfile_yaml):
     df.set_index('antname', inplace=True)
 
     return df
+
+
+def update_antpos_etcd():
```
Comment (yupinghuang) [inline lwa_antpos/reading.py:75]: um, are you sure you want everyone who imports this module to be able to change the etcd thing for everyone?

**Review by yupinghuang — CHANGES_REQUESTED** [review]
So, https://github.com/ovro-lwa/lwa-antpos/blob/main/lwa_antpos/__init__.py does some of this too. You probably want to make up your mind whether you want people to do `from lwa_antpos.reading import read_antpos` or `from lwa_antpos import lwa_df` and ask them to stick to that to avoid nightmares down the road.

I think this is a good place for abstraction. Ultimately, the user wants to get the current data frame with columns that they expect. Very few people would care where they get the information from, unless the source they're using is broken, which is where abstraction with this interface comes in.

File: lwa_antpos/reading.py (hunk 1 above)
Comment (caseyjlaw) [reply lwa_antpos/reading.py]: agreed

File: lwa_antpos/reading.py (hunk 2 above)
Comment (caseyjlaw) [reply lwa_antpos/reading.py:75]: Fair point. This is a convenience function, but setting etcd probably shouldn't be too convenient. I'll punt on this for now and note that we need to define controls on setting to etcd.

**Comment by caseyjlaw** [comment]
Thanks for the review. I'll merge now.

**Pull request merged**